### ISSUE TITLE
Trouble getting Pojo from (Composite) index through Repository

### ISSUE THREAD

**Issue by sblommers** [xvik/guice-persist-orient#12]
Hi there, I'm pretty new with this library and using OrientDB but I am stuck on selecting some entities from an index (in my case composite but i have same result on single/multiple). 
The class at: https://github.com/xvik/guice-persist-orient/blob/ece1980d4dff42fac1d74d46a70053862a8e2f84/src/test/groovy/ru/vyarus/guice/persist/orient/study/index/ci/IndexCiRepository.groovy triggered my attention that it should be possible to use the index to fetch the entity directly. Unfortunately upon doing so I keep getting:

java.lang.ClassCastException: 
**com.orientechnologies.orient.core.record.impl.ODocument cannot be cast to orientdb.test.index.IndexTest$TestEntity**

My query on the repository is:
```java
@Query("select from index:t where key=[?,?]")
```

I am wondering if i am doing something wrong.
Here is my TestNG test reproducing this problem for me:
```java
package orientdb.test.index;

import javax.inject.Inject;
import javax.inject.Singleton;

import org.testng.annotations.BeforeTest;
import org.testng.annotations.Test;

import com.google.inject.AbstractModule;
import com.google.inject.Guice;
import com.google.inject.ProvidedBy;
import com.google.inject.internal.DynamicSingletonProvider;
import com.google.inject.persist.PersistService;
import com.google.inject.persist.Transactional;
import com.orientechnologies.orient.core.metadata.schema.OClass.INDEX_TYPE;

import ru.vyarus.guice.persist.orient.OrientModule;
import ru.vyarus.guice.persist.orient.RepositoryModule;
import ru.vyarus.guice.persist.orient.db.scheme.annotation.Persistent;
import ru.vyarus.guice.persist.orient.db.scheme.initializer.ext.type.index.CompositeIndex;
import ru.vyarus.guice.persist.orient.model.VersionedEntity;
import ru.vyarus.guice.persist.orient.repository.command.query.Query;
import ru.vyarus.guice.persist.orient.support.AutoScanSchemeModule;
import ru.vyarus.guice.persist.orient.support.repository.mixin.crud.ObjectCrud;

public class IndexTest {
	
	@BeforeTest
	public void init() throws Exception {		
		Guice.createInjector(new AbstractModule() {
			@Override
			protected void configure() {
				install(new OrientModule("memory:test", "admin", "admin"));
				install(new AutoScanSchemeModule(this.getClass().getPackage().getName()));
				install(new RepositoryModule());
				bind(Initializer.class).asEagerSingleton();
			}
		}).injectMembers(this);
	}
	
	@Inject TestEntityRepository testRepo;
	
	@Test
	public void run() throws Exception {
		final TestEntity te = new TestEntity();
		te.id1 = 1;
		te.id2 = 2;
		testRepo.save(te);
		
		testRepo.loadIndex(1,2);
	}
	
	@Persistent	@CompositeIndex(name="t", fields={"id1", "id2" }, type = INDEX_TYPE.NOTUNIQUE)
	class TestEntity extends VersionedEntity {
		long id1;
		long id2;
	}
	
	@Transactional @ProvidedBy(DynamicSingletonProvider.class)
	public interface TestEntityRepository extends ObjectCrud<TestEntity> {
		@Query("select from index:t where key=[?,?]")
		TestEntity loadIndex(long id1, long id2);
	}
	
	@Singleton
	public static class Initializer {
		@Inject
		public Initializer(PersistService service) {
			service.start();
		}
	}
}
```
Basically I have TestEntity with id1 and id2 that are a composite index as well. I use TestEntityRepository to save entity and use loadIndex using the composite key to fetch the TestEntity. Unfortunately the TestEntityRepository generated loadIndex returns a ClassCastException because ODocument cannot be cast to TestEntity.

What am I missing?

Thanks for your help and this great guice integration

**Comment by xvik** [comment]
Hi,

My test was incorrect: when you query index you actually receive key-value (key - rid) pair which could be represented only as Document. 

Use this query to return complete object (which will be mapped to your pojo):
```
select expand(rid) from index:t where key=[?,?]
```

**Comment by sblommers** [comment]
Thank you for your quick reply, this fixes my issue :+1:
